tls/s2n_x509_validator.c

```diff
@@ -284,16 +286,16 @@ static uint8_t s2n_verify_host_information(struct s2n_x509_validator *validator,
     return verified;
 }
 
-s2n_cert_validation_code s2n_x509_validator_validate_cert_chain(struct s2n_x509_validator *validator, struct s2n_connection *conn,
+S2N_RESULT s2n_x509_validator_validate_cert_chain(struct s2n_x509_validator *validator, struct s2n_connection *conn,
         uint8_t *cert_chain_in, uint32_t cert_chain_len, s2n_pkey_type *pkey_type, struct s2n_pkey *public_key_out) {
-    S2N_ERROR_IF(!validator->skip_cert_validation && !s2n_x509_trust_store_has_certs(validator->trust_store), S2N_ERR_CERT_UNTRUSTED);
-    S2N_ERROR_IF(validator->state != INIT, S2N_ERR_INVALID_CERT_STATE);
+    RESULT_ENSURE(validator->skip_cert_validation || s2n_x509_trust_store_has_certs(validator->trust_store), S2N_ERR_CERT_UNTRUSTED);
+    RESULT_ENSURE(validator->state == INIT, S2N_ERR_INVALID_STATE);
 
```

Comment:
Why change this error? S2N_ERR_INVALID_CERT_STATE exists specifically for this use case, and is an internal error instead of a usage error like S2N_ERR_INVALID_STATE.

Reply:
Sorry, this was not intentional. I fixed it in b51772769dee845b2983e520943b84b3335b083d